### ISSUE TITLE
Qt Error on cv2.imshow - Specific to version 4.2.0.32

### ISSUE THREAD

**Issue by NeilGraham** [skvark/opencv-python#290]
### Issue
Calling `cv2.imshow` with `opencv-python 4.2.0.32` results in the following error:
```
qt.qpa.plugin: Could not find the Qt platform plugin "cocoa" in ""
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.
```
### Solution
I am able to fix the issue by:
- uninstalling the current version: `4.2.0.32`
- reinstalling the previous version: `4.1.2.30`
```
pip uninstall opencv-python
pip install opencv-python==4.1.2.30
```

### Repoduce

```python
import cv2 
img = cv2.imread('./path/to/picture.png', cv2.IMREAD_GRAYSCALE)
cv2.imshow('image', img)
```

- macOS 10.15.2 x86
- Python 3.8.1
- opencv-python 4.2.0.32

**Comment by skvark** [comment]
Qt version was upgraded from 4 to 5 in the latest update (Qt 4 dependencies were too old and caused major issues during builds). The issue is most likely related to that upgrade. I don't own a Mac so it's quite hard for me to debug the issue. However, I'll try to look into it.

**Comment by skvark** [comment]
See: https://forum.qt.io/topic/98816/qt-could-not-find-the-platform-plugin-cocoa

Set ``QT_DEBUG_PLUGINS`` to 1 before running your script. It will tell where Qt is looking for the cocoa plugin.

**Comment by dlech** [comment]
This is with `opencv-contrib-python` 4.2.0.32 in a virtual environment.

```
QFactoryLoader::QFactoryLoader() checking directory path "/usr/local/Cellar/python@3.8/3.8.1/Frameworks/Python.framework/Versions/3.8/Resources/Python.app/Contents/MacOS/platforms" ...
qt.qpa.plugin: Could not find the Qt platform plugin "cocoa" in ""
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

zsh: abort      QT_DEBUG_PLUGINS=1 python __main__.py
```

I made some progress setting `QT_QPA_PLATFORM_PLUGIN_PATH` to the Qt path for the homebrew install of Qt. It gives errors about not being compatible though since it is not the same build as the Qt library in the wheel.

There is `/usr/local/Cellar/qt/5.13.1/plugins/platforms/libqcocoa.dylib` installed by homebrew, but no such plugin in the opencv-contrib-python wheel.

So, I think the first step is to get `libqcocoa.dylib` in the binary wheel.

**Comment by NeilGraham** [comment]
I am getting a similar error message:
```bash
export QT_DEBUG_PLUGINS=1
python3 run.py
```
```
QFactoryLoader::QFactoryLoader() checking directory path "/Library/Frameworks/Python.framework/Versions/3.8/Resources/Python.app/Contents/MacOS/platforms" ...
qt.qpa.plugin: Could not find the Qt platform plugin "cocoa" in ""
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

zsh: abort      python3 run.py
```
- For what it's worth, I did have an independent version of Qt 5 installed on my computer when I encountered this issue. I attempted to reinstall Qt to fix the issue but had no luck (same error message). At the moment, it is not installed on my system as of executing the command above.

**Comment by skvark** [comment]
@dlech Yeah, it looks like ``libqcocoa.dylib`` needs to be embedded to the wheel. Also ``qt.conf`` file is maybe needed to set the plugin search path relative to the wheel binaries folder: https://doc.qt.io/qt-5/qt-conf.html

I don't own a Mac so it would be helpful if you can test if the Qt which is embedded to the wheels can be configured to look for the dylib from the wheel install directory.

**Comment by dlech** [comment]
The `QT_QPA_PLATFORM_PLUGIN_PATH` environment variable can be used to set the plugin search path.

So if we can make it so that `QT_QPA_PLATFORM_PLUGIN_PATH` is set when the `cv2` module is imported, that could be a solution.

**Comment by skvark** [comment]
I'm pretty sure that won't work because the environment variable must be set outside / before the process.

**Comment by dlech** [comment]
It should be fine as long as it is set before we try to use Qt.

**Comment by skvark** [comment]
It's not very convenient to require every user to export that environment variable before use.

**Comment by dlech** [comment]
Yes. I am suggesting that we do it automatically when the `cv2` module is imported so that users don't have to do it themselves.

**Comment by dlech** [comment]
I spend quite a few hours last night trying to get this repo to build so I can actually try it out. I never could get it to build with python3, but finally got it to build with python2. So making some progress.

**Comment by skvark** [comment]
It's quite hard to do the export because it must be done outside the Python process. You can't just set it with Python because the process has already started and afaik setting env variables during the process does not have any effect.

**Comment by dlech** [comment]
Please try this simple example to see that it is in fact possible to modify environment variables after a process starts. It is true that other processes cannot modify the variable, but the process itself can.

```python
import os

os.environ['PYTHONINSPECT'] = 'on'
```

It has the same effect as `PYTHONINSPECT=on python`

**Comment by skvark** [comment]
Thanks for the clarification, I probably mixed up this with the case where another process needed to modify the environment variables (I was trying to do something like that a few years back with some other project...). Thanks for the PR, I'll have a look at it during next week.

**Comment by justinkterry** [comment]
When will this be fixed? This is a real problem for me

**Comment by prasadnoolu** [comment]
try degrading opencv-contrib by something like sudo pip install opencv-contrib-python==4.0.0.21

this will help

**Comment by skvark** [comment]
@prasadnoolu This has been fixed and released in 4.2.0.34.
